src/operations/blas3/gemm_partial_local.hpp

```diff
@@ -0,0 +1,454 @@
+                                    global_n_offset, global_k_offset);
+    else
+      load_blocks<false, false, false>(local_id, tile_idx, scratch_ptr,
+                                       rhs_scratch_ptr, global_m_offset,
+                                       global_n_offset, global_k_offset);
+  }
+
+  template <bool check_m_limit, bool check_n_limit, bool check_k_limit,
+            typename local_ptr_t>
+  SYCL_BLAS_INLINE void load_blocks(index_t local_id, index_t tile_idx,
+                                    local_ptr_t scratch_ptr,
+                                    local_ptr_t rhs_scratch_ptr,
+                                    index_t global_m_offset,
+                                    index_t global_n_offset,
+                                    index_t global_k_offset) {
+    // LHS tile
+    if (trans_a) {
+      load_and_transpose_block<loads_per_thread_lhs, check_k_limit,
+                               check_m_limit, tile_size_dim_k, tile_size_dim_m,
+                               ld_lhs_tile>(local_id, tile_idx, a_, lda_,
+                                            scratch_ptr, global_k_offset,
+                                            global_m_offset, k_, m_);
+    } else {
+      load_block<loads_per_thread_lhs, check_m_limit, check_k_limit,
+                 tile_size_dim_m, tile_size_dim_k, ld_lhs_tile>(
+          local_id, tile_idx, a_, lda_, scratch_ptr, global_m_offset,
+          global_k_offset, m_, k_);
+    }
+    // RHS tile
+    if (trans_b) {
+      load_block<loads_per_thread_rhs, check_n_limit, check_k_limit,
+                 tile_size_dim_n, tile_size_dim_k, ld_rhs_tile>(
+          local_id, tile_idx, b_, ldb_, rhs_scratch_ptr, global_n_offset,
+          global_k_offset, n_, k_);
+    } else {
+      load_and_transpose_block<loads_per_thread_rhs, check_k_limit,
+                               check_n_limit, tile_size_dim_k, tile_size_dim_n,
+                               ld_rhs_tile>(local_id, tile_idx, b_, ldb_,
+                                            rhs_scratch_ptr, global_k_offset,
+                                            global_n_offset, k_, n_);
+    }
+  }
+
+  template <index_t loads_per_thread, bool check_row_limit,
+            bool check_col_limit, index_t block_rows, index_t block_cols,
+            index_t ld_tile, typename local_ptr_t>
+  static SYCL_BLAS_INLINE void load_block(
+      index_t local_id, index_t tile_idx, input_t& in_view, index_t leading_dim,
+      local_ptr_t local_ptr, index_t global_row_offset,
+      index_t global_col_offset, index_t global_rows, index_t global_cols) {
+    index_t local_linear_id = local_id;
+    const index_t global_tile_col_offset =
+        global_col_offset + block_cols * tile_idx;
+    const index_t local_thread_size = local_thread_size_n * local_thread_size_m;
+
+    // Double buffering
+    constexpr index_t block_size = ld_tile * block_cols;
+    const index_t local_mem_offset = (tile_idx & 1) * block_size;
+
```

Comment:
#pragma unroll

Reply:
Added, I don't see any performance gain but there is no regression either